### PR TITLE
Flesh out DOM interaction helper documentation.

### PR DESCRIPTION
**Pull request by rwjblue**

### CODE REVIEW

File: addon-test-support/@ember/test-helpers/dom/click.js [hunk 1]
```diff
@@ -22,10 +17,12 @@ export function __click__(element) {
 }
 
 /**
-  @method click
-  @param {String|Element} target
-  @return {Promise<void>}
+  Clicks on the specified selector.
```
Comment (Turbo87) [inline addon-test-support/@ember/test-helpers/dom/click.js]: "specified target" might be better

File: addon-test-support/@ember/test-helpers/dom/fire-event.js [hunk 2]
```diff
@@ -15,15 +15,6 @@ const MOUSE_EVENT_TYPES = [
...
 
-/**
-  @method fireEvent
-  @param {Element} element
-  @param {String} type
-  @param {Object} [options]
-  @returns {Event}
-
-  @private
-*/
```
Comment (Turbo87) [inline addon-test-support/@ember/test-helpers/dom/fire-event.js]: Why did you remove these? They might still be useful for type info

File: addon-test-support/@ember/test-helpers/dom/trigger-key-event.js [hunk 3]
```diff
@@ -13,16 +13,18 @@ const DEFAULT_MODIFIERS = Object.freeze({
 });
 
 /**
+  Triggers a keyboad event on the specified selector.
```
Comment (Turbo87) [inline addon-test-support/@ember/test-helpers/dom/trigger-key-event.js]: Keyboard

File: addon-test-support/@ember/test-helpers/dom/fire-event.js (hunk 2 above)
Comment (rwjblue) [reply addon-test-support/@ember/test-helpers/dom/fire-event.js]: Hmm, unless we are going to type check ember-test-helpers internally, I don't think this provides any value. This entire file is private and only used to expose the actual public API's which _do_ have valid type information...

Comment (Turbo87) [reply addon-test-support/@ember/test-helpers/dom/fire-event.js]: Some IDEs are able to parse these comments and provide hints. Might be useful for future contributors.

Comment (rwjblue) [reply addon-test-support/@ember/test-helpers/dom/fire-event.js]: OK, will bring it back for the export (but not the inner / private helper functions)
